src/main/java/pro/belbix/ethparser/codegen/SimpleContractGenerator.java

```diff
@@ -53,27 +56,41 @@
   private final Web3Functions web3Functions;
   private final NetworkProperties networkProperties;
   private final ContractDbService contractDbService;
+  private final ContractSourceCodeRepository contractSourceCodeRepository;
+  private final AppProperties appProperties;
 
   private final Map<String, TreeMap<Long, GeneratedContract>> contracts = new HashMap<>();
 
   public SimpleContractGenerator(FunctionsUtils functionsUtils,
       Web3Functions web3Functions,
       NetworkProperties networkProperties,
-      ContractDbService contractDbService) {
+      ContractDbService contractDbService,
+      ContractSourceCodeRepository contractSourceCodeRepository,
+      AppProperties appProperties
+      ) {
     this.functionsUtils = functionsUtils;
     this.web3Functions = web3Functions;
     this.networkProperties = networkProperties;
     this.contractDbService = contractDbService;
     this.abiProviderService = new AbiProviderService();
+    this.contractSourceCodeRepository = contractSourceCodeRepository;
+    this.appProperties = appProperties;
   }
 
   public GeneratedContract getContract(String address, Long block, String selector,
       String network) {
+
+    Boolean isUseCache = appProperties.isUseContractSourceCache();
```

Comment:
I guess we can use it without adding as attribute for each method

Reply:
ACK